subsys/net/ip/tcp.c

```diff
@@ -2430,19 +2430,41 @@ static enum net_verdict tcp_in(struct tcp *conn, struct net_pkt *pkt)
 		} else if (th && FL(&fl, ==, ACK, th_seq(th) == conn->ack)) {
 			tcp_send_timer_cancel(conn);
 			next = TCP_FIN_WAIT_2;
+		} else if (th && (len > 0)) {
+			tcp_send_timer_cancel(conn);
```

Comment:
Why could we not use all the changes from TCP_FIN_WAIT_1 state? Is there some logic that's incompatible with our version?

Reply:
@kica-z  
First, this  is from another PR `verdict = NET_OK;` (but this is only for statistics, if my understanding is correct, see 

Also, I cherry-picked only what was addressing our problem and what I understood, being afraid of side-effects.
Actually, before finding that PR, I partly implemented the fix myself, which was similar.

But in general you are right, we can cherry-pick the change completely.
I don't have a strong opinion here.

Reply:
i think especially with the new update in mind we should keep the merge as simple as possible. If we cherry-pick the entire change there will be no conflict later on.
But I think the current version also works fine.

@clamattia what do you think?

Reply:
I tend to agree, but please note that this particular part will be again replaced in https://github.com/zephyrproject-rtos/zephyr/pull/61353 (but it comes after 3.4.0)